datafusion/src/sql/planner.rs

```diff
@@ -213,9 +213,19 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
     ) -> Result<LogicalPlan> {
         let set_expr = &query.body;
         if let Some(with) = &query.with {
+            // a `WITH` block can't use the same name for many times
+            let mut cte_names = HashSet::new();
             // Process CTEs from top to bottom
             // do not allow self-references
             for cte in &with.cte_tables {
+                let cte_name = cte.alias.name.value.clone();
```

Comment:
I think you can avoid cloing the string here:

```suggestion
                let cte_name = cte.alias.name.value.as_ref();
```